test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -832,6 +832,33 @@ public async Task DotnetRestore_LockedMode_NewProjectOutOfBox()
             }
         }
 
+        [PlatformFact(Platform.Windows)]
+        public void DotnetRestore_LockedMode_Net5WithAndWithoutPlatform()
+        {
+            using (var pathContext = _msbuildFixture.CreateSimpleTestPathContext())
+            {
+                // Arrange
+                string projectFileContents =
+@"<Project Sdk=""Microsoft.NET.Sdk"">
+    <PropertyGroup>
+        <TargetFrameworks>net5.0;net5.0-windows</TargetFrameworks>
+        <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
+    </PropertyGroup>
+</Project>";
+                File.WriteAllText(Path.Combine(pathContext.SolutionRoot, "a.csproj"), projectFileContents);
+
+                // Act
+                _msbuildFixture.RestoreProject(pathContext.SolutionRoot, "a", args: null);
```

Comment:
Can you rerun restore with the cache deleted + locked mode?

As it is right now, not sure the equality codepath is exercised.